### PR TITLE
#170947382 Users should be able to Login via Facebook and Google. 

### PR DESCRIPTION
**Pull request by blaise82**
#### What does this PR do?
- Users should be able to Login via Facebook and Google. 

#### Description of Task to be completed?
- add click to social buttons
- handle click on social login
- create callback component

#### How should this be manually tested?
- npm close this repo `git clonehttps://github.com/Stackup-Rwanda/octopus-bn-frontend`
- check out to this branch `ft-social-login-170947382`
- npm install all required dependencies  `npm install`
- start client-server `npm run dev`
- if your testing this locally please create a .env file in the parent directory 
and those two variables
```
FACEBOOK_SERVER_CALL=http://localhost:3000/api/v1/auth/facebook
GOOGLE_SERVER_CALL=http://localhost:3000/api/v1/auth/google

```
Note: Make sure that your backend server is on

#### Screenshots (if appropriate)

screencapture-accounts-google-signin-oauth-oauthchooseaccount-2020-04-28-12_21_49

[Finishes #170947382]"

### CODE REVIEW

**Comment by ceelogre** [comment]
Create a review app.

**Comment by blaise82** [comment]
> Create a review app.

working on it

**Review by AbdoulNuru — DISMISSED** [review]

File: src/views/Login.View.js [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
 export default class Login extends Component {
+	constructor(props) {
+		super(props);
+    }
 	render() {
+		const token = localStorage.getItem('bn-token');
+		const user = localStorage.getItem('bn-user-data');
+		if(token){
+			return <Redirect to='/dashboard' />
+		}
```
Comment (AJAkimana) [inline src/views/Login.View.js:17]: Put this check in main component

File: src/views/SignupPage.js [hunk 2]
```diff
@@ -13,6 +14,11 @@ class SignupPage extends React.Component {
       .then(() => this.props.history.push("/verify-email"));
 
   render() {
+    const token = localStorage.getItem('bn-token');
+		const user = localStorage.getItem('bn-user-data');
+		if(token){
+			return <Redirect to='/dashboard' />
+		}
```
Comment (AJAkimana) [inline src/views/SignupPage.js:21]: Make these check-ins in the main component

File: src/helpers/SocialLogin.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+		super(props);
+	}
+	componentDidMount() {
+		try {
+			const values = queryString.parse(this.props.location.search);
+			const data = values.data;
+			const user = JSON.parse(data)[0];
+			const token = JSON.parse(data)[2];
+			localStorage.setItem('bn-token', token);
+			localStorage.setItem('bn-user-data', JSON.stringify(user));
```
Comment (AJAkimana) [inline src/helpers/SocialLogin.js:16]: Save use info in the state(`redux`) so that you can access whenever you want

**Review by Moise1 — DISMISSED** [review]

**Review by AbdoulNuru — DISMISSED** [review]

**Review by hakoemmy — DISMISSED** [review]

**Review by AJAkimana — COMMENTED** [review]
You need to check the expiration of the token then decide what to do the next(Redirecting to log in, dashboard, or home page). This will also take care for persisting login state after refreshing

**Review by hakoemmy — APPROVED** [review]

**Review by AbdoulNuru — APPROVED** [review]

**Pull request merged**